### PR TITLE
merge develop -> master

### PR DESCRIPTION
**Pull request by JuusoMRusanen**
Merging master with develop! Let's check this code for any strange stuff before merging.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      <ThemeProvider theme={theme}>
+        <ErrorHandler>
+          <AuthenticationProvider>
+            <CssBaseline>
+              <RouterProvider router={router} />
+            </CssBaseline>
+          </AuthenticationProvider>
+        </ErrorHandler>
+      </ThemeProvider>
+    </div>
```
Comment (petri-nykanen) [inline src/App.tsx:41]: Localization provider is missing from this version.

File: src/components/screens/home/balance-card.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import { PersonTotalTime } from "../../../generated/client";
+
+/**
+ * Component for displaying user's balance
+ */
+const BalanceCard = () => {
+  const userProfile = useAtomValue(userProfileAtom);
+  const { personsApi } = useApi();
+  const setError = useSetAtom(errorAtom);
+  const [isLoading, setIsLoading] = useState(false);
```
Comment (anttileppa) [inline src/components/screens/home/balance-card.tsx:19]: verbs should be used only in method

```suggestion
  const [loading, setLoading] = useState(false);
```

File: src/components/screens/home/balance-card.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  const [personTotalTime, setPersonTotalTime] = useState<PersonTotalTime>();
+
+  /**
+   * Initialize logged in person's time data.
+   */
+  const getPersons = async () => {
+    setIsLoading(true);
+    const fetchedPersons = await personsApi.listPersons({ active: true });
+    const loggedInPerson = fetchedPersons.filter((person) => person.keycloakId === userProfile?.id);
```
Comment (anttileppa) [inline src/components/screens/home/balance-card.tsx:28]: Replace with find

File: src/components/screens/home/balance-card.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      setPersonTotalTime(fetchedPerson[0]);
+    } catch (error) {
+      setError(`${strings.errors.fetchFailedGeneral}, ${error}`);
+    }
+
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
+    getPersons();
```
Comment (anttileppa) [inline src/components/screens/home/balance-card.tsx:43]: Add error handling

File: src/utils/time-utils.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+/**
+ * Converts inputted minutes into hours and minutes
+ * @param minutes value in minutes
+ * @returns inputted minute value in X h Y min format as string
+ */
+export const getHoursAndMinutes = (minutes : number) : string => {
+    return `${Math.trunc(minutes / 60)} h ${(minutes % 60) * (String(minutes).startsWith("-")?-1 :1)} min`
```
Comment (anttileppa) [inline src/utils/time-utils.ts:7]: Replace with proper date formatter (https://github.com/moment/luxon/blob/master/docs/formatting.md)

**Review by anttileppa — CHANGES_REQUESTED** [review]

**Review by samisii — APPROVED** [review]

**Review by anttileppa — APPROVED** [review]

**Pull request merged**